plugin/src/main/java/org/opensearch/ml/action/register/TransportRegisterModelAction.java

```diff
@@ -239,7 +240,14 @@ private void doRegister(MLRegisterModelInput registerModelInput, ActionListener<
             if (Strings.isNotBlank(registerModelInput.getConnectorId())) {
                 connectorAccessControlHelper.validateConnectorAccess(client, registerModelInput.getConnectorId(), ActionListener.wrap(r -> {
                     if (Boolean.TRUE.equals(r)) {
-                        createModelGroup(registerModelInput, listener);
+                        if (registerModelInput.getModelInterface() == null) {
```

Comment:
are we expecting a new field in model index? I don't see any declaration of this new field? Or did you already do that in any other PR?